src/UCTNode.h

```diff
@@ -113,7 +195,71 @@ class UCTNode {
 
     // Tree data
     std::atomic<bool> m_has_children{false};
-    std::vector<node_ptr_t> m_children;
+    std::vector<UCTNodePointer> m_children;
 };
 
+
+inline UCTNodePointer::~UCTNodePointer() {
```

Comment:
Is there any particular reason why these are all inline in the header? I'd prefer all the UCTNodePointer things in a separate .h/.cpp in fact.

Reply:
Well, it's an old habit dealing with older (e.g., gcc 4.7) compilers.  Experience is that when it's only small snippets and it's a critical piece of code, just putting it into headers yield much better performance.  If you think modern linkers are much better at handling global optimization I see no reason to keep them in the header though.  If you think performance won't change much then I have no reason that we keep in the header.